Source/DataModels/Windows/WindowMainMenu.cpp

```diff
@@ -27,51 +30,129 @@ void WindowMainMenu::Draw(bool& enabled)
 {
 	if (ImGui::BeginMainMenuBar())
 	{
-		DrawWindowsMenu();
-		DrawAbout();
-		DrawGithubLink();
-		DrawExit();
+		DrawFileMenu();
+		DrawWindowMenu();
+		DrawHelpMenu();
+		if (openPopup) DrawPopup();
+		if (isSaving) saveScene->SaveAsWindow(isSaving);
 	}
 	ImGui::EndMainMenuBar();
 }
 
-void WindowMainMenu::DrawWindowsMenu()
+void WindowMainMenu::DrawPopup()
 {
-	if (ImGui::BeginMenu("Windows"))
+	ImGui::OpenPopup("Are you sure?");
+	ImVec2 center = ImGui::GetMainViewport()->GetCenter();
+	ImGui::SetNextWindowPos(center, ImGuiCond_Appearing, ImVec2(0.5f, 0.5f));
+	if (ImGui::BeginPopupModal("Are you sure?", NULL, ImGuiWindowFlags_AlwaysAutoResize))
 	{
-		for (std::pair<std::string, bool>& windowNameAndEnabled : windowNamesAndEnabled)
+		ImGui::Text("Do you want to save the scene?\nAll your changes will be lost if you don't save them.");
+		ImGui::Separator();
+
+		if (ImGui::Button("Save scene", ImVec2(120, 0)))
 		{
-			ImGui::MenuItem(windowNameAndEnabled.first.c_str(), NULL, &windowNameAndEnabled.second);
+			std::string filePathName = App->scene->GetLoadedScene()->GetRoot()->GetName();
+			if (filePathName != "New Scene")
+			{
+				App->scene->SaveSceneToJson(filePathName + ".axolotl");
```

Comment:
```suggestion
App->scene->SaveSceneToJson(filePathName + SCENE_EXTENSION);
```

Prefer using the global `SCENE_EXTENSION` instead of `".axolotl"`.